utils/providers.py

```diff
@@ -31,56 +20,26 @@
 from cfme.middleware.provider import HawkularProvider
 from fixtures.prov_filter import filtered
 from utils import conf, version
-from utils.mgmt_system import RHEVMSystem
 from utils.log import logger, perflog
 from utils.wait import wait_for
 
-#: mapping of infra provider type names to ``mgmtsystem`` classes
-infra_provider_type_map = {
-    'virtualcenter': VMWareSystem,
-    'rhevm': RHEVMSystem,
-    'scvmm': SCVMMSystem,
-    'openstack-infra': OpenstackInfraSystem,
-}
-
-#: mapping of cloud provider type names to ``mgmtsystem`` classes
-cloud_provider_type_map = {
-    'ec2': EC2System,
-    'openstack': OpenstackSystem,
-    'gce': GoogleCloudSystem,
-}
-
-#: mapping of container provider type names to ``mgmtsystem`` classes
-container_provider_type_map = {
-    'kubernetes': Kubernetes,
-    'openshift': Openshift
-}
-
-#: mapping of middleware provider type names to ``mgmtsystem`` classes
-middleware_provider_type_map = {
-    'hawkular': Hawkular,
-}
-
-
-#: mapping of all provider type names to ``mgmtsystem`` classes
-provider_type_map = {}
-provider_type_map.update(infra_provider_type_map)
-provider_type_map.update(cloud_provider_type_map)
-provider_type_map.update(container_provider_type_map)
-provider_type_map.update(middleware_provider_type_map)
 
 providers_data = conf.cfme_data.get("management_systems", {})
 
 # This is a global variable. Not the most ideal way to do things but how can we track bad providers?
 problematic_providers = set([])
 
 
-def list_providers(allowed_types):
+def list_providers(allowed_types=None):
     """ Returns list of providers of selected type from configuration.
 
     @param allowed_types: Passed by partial(), see top of this file.
     @type allowed_types: dict, list, set, tuple
     """
+    if not allowed_types:
+        allowed_types = [
+            k2 for k in BaseProvider.type_mapping.keys() for k2 in BaseProvider.type_mapping[k]
```

Comment:
could be golfed into `set(itertools.chain.from_iterable(BaseProvider.type_mapping.values()))` :smile: